### PR TITLE
Use main react-transition group

### PR DESCRIPTION
**Pull request by BPScott**
WIP - Still need to test this

### WHY are these changes introduced?

We started using a fork of react-transition-group because the original didn't support strict mode.

Now the original supports strict-mode so let's start using it again.

### WHAT is this pull request doing?

Migrate from `@material-ui/react-transition-group` to `react-transition-group`

### CODE REVIEW

**Comment by nicolad** [comment]
out of curiosity: why not `react-spring` ? 
one big advantage is hooks API

File: src/components/Frame/Frame.tsx [hunk 1]
```diff
@@ -135,7 +135,7 @@ class FrameInner extends React.PureComponent<CombinedProps, State> {
     const navigationMarkup = navigation ? (
       <TrapFocus trapping={mobileNavShowing}>
         <CSSTransition
-          findDOMNode={this.findNavigationNode}
+          nodeRef={this.navigationNode}
```
Comment (AndrewMusgrave) [inline src/components/Frame/Frame.tsx:138]: Glad they made the API accept refs. Wasn't a fan of the function approach

**Comment by AndrewMusgrave** [comment]
> out of curiosity: why not react-spring ?

This pull request isn't adding `react-transition-group`. It migrating from a forked version 😄

**Comment by dleroux** [comment]
> out of curiosity: why not `react-spring` ?
> one big advantage is hooks API

We explored it a while back, but felt it wasn't worth the file size to add to Polaris at the time.

**Pull request merged**